src/gui/src/layoutViewer.cpp

```diff
@@ -1983,6 +1992,46 @@ void LayoutViewer::drawBlock(QPainter* painter,
   }
 }
 
+void LayoutViewer::drawRegionOutlines(QPainter* painter)
+{
+  if (!options_->areRegionsVisible()) {
+    return;
+  }
+
+  painter->setPen(QPen(Qt::gray, 0));
+  painter->setBrush(Qt::BrushStyle::NoBrush);
+
+  for (auto* region : block_->getRegions()) {
+    const odb::Rect region_box = getRegionRect(region);
+    if (region_box.area() > 0) {
+      painter->drawRect(region_box.xMin(), region_box.yMin(), region_box.dx(), region_box.dy());
+    }
+  }
+}
+
+odb::Rect LayoutViewer::getRegionRect(odb::dbRegion* region) const
+{
+  if (region == nullptr) {
+    return {};
+  }
+
+  auto boundaries = region->getBoundaries();
+  if (boundaries.empty()) {
+    return getRegionRect(region->getParent());
+  } else {
+    odb::Rect region_box;
+    region_box.mergeInit();
+
+    for (auto* box : boundaries) {
+      odb::Rect box_rect;
+      box->getBox(box_rect);
+      region_box.merge(box_rect);
+    }
```

Comment:
You should draw each rect of the region not their union (imagine an L shaped region).  If you are just drawing the outline you'll need to build a polygon and extract the boundary.

Reply:
Just drawing the rects makes sense without merging